src/transformers/models/greaselm/feature_extraction_greaselm.py

```diff
@@ -0,0 +1,739 @@
+        cpnet_vocab_path: Union[Path, str] = "concept.txt",
+        pattern_path: Union[Path, str] = "matcher_patterns.json",
+        pruned_graph_path: Union[Path, str] = "conceptnet_en_pruned.pickle.gz",
+        score_model: Union[Path, str] = "roberta-large",
+        device: str = "cuda",
+        cxt_node_connects_all: bool = False,
+        **kwargs
+    ):
+        requires_backends(self, ["spacy", "scatter", "sparse"])
+        super().__init__(**kwargs)
+        self.cpnet_vocab_path = cpnet_vocab_path
+        self.pattern_path = pattern_path
+        self.pruned_graph_path = pruned_graph_path
+        self.score_model = score_model
+        self.device = device
+
+        self.nlp = None
+        self.matcher = None
+        self.tokenizer = None
+        self.model = None
+        self.loss_fct = None
+        self.cxt_node_connects_all = cxt_node_connects_all
+
+        self.cpnet_vocab = None
+        self.cpnet_vocab_underscores = None
+        self.concept2id = None
+        self.id2relation = None
+        self.id2concept = None
+        self.relation2id = None
+
+        self.cpnet_simple = None
+        self.cpnet = None
+        self.started = False
+
+    def __call__(
+        self,
+        question_answer_example: Union[Dict[str, Any], List[Dict[str, Any]]],
+        entailed_question_answer_example: Union[Dict[str, Any], List[Dict[str, Any]]],
+        num_choices: int = 5,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        **kwargs
+    ) -> Dict[str, Any]:
+        """
+        Main method to encode a question-answer example(s) into a graph representation ready for model input.
+
+        Args:
+            question_answer_example (`List[Dict[str, Any]]`):
+                A question-answer example or a batch of question-answer examples from CommonSenseQA or OpenBookQA
+                datasets.
+
+            entailed_question_answer_example (`List[Dict[str, Any]]`):
+                An entailed question-answer example or a batch of question-answer examples from CommonSenseQA or
+                OpenBookQA datasets.
+
+            num_choices (`int`, *optional*, defaults to `5`):
+                Number of choices to in the input example
+
+            return_tensors (`str` or [`~utils.TensorType`], *optional*, defaults to `'pt'`):
+                Currently, only pt is supported.
+
+        Returns:
+            [`dict`]: A [`dict`] with the following fields:
+
+            - concept_ids: (batch_size, num_choices, max_node_num)
+            - node_type_ids: (batch_size, num_choices, max_node_num)
+            - node_scores: (batch_size, num_choices, max_node_num, 1)
+            - adj_lengths: (batch_size, num_choices)
+            - special_nodes_mask: (batch_size, num_choices, max_node_num)
+            - edge_index: list of size (batch_size, num_choices), where each entry is tensor[2, E]
+            - edge_type: list of size (batch_size, num_choices), where each entry is tensor[E, ]
```

Comment:
```suggestion
            - `concept_ids`(`torch.FloatTensor` of shape `(batch_size, num_choices, max_node_num)`):
                A tensor containing (...).
            - node_type_ids: (batch_size, num_choices, max_node_num)
            - node_scores: (batch_size, num_choices, max_node_num, 1)
            - adj_lengths: (batch_size, num_choices)
            - special_nodes_mask: (batch_size, num_choices, max_node_num)
            - edge_index: list of size (batch_size, num_choices), where each entry is tensor[2, E]
            - edge_type: list of size (batch_size, num_choices), where each entry is tensor[E, ]
```
Please follow the docs format, as explained [here](https://github.com/huggingface/transformers/tree/main/docs#writing-documentation---specification).